language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/CommandsBuilder.java

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.rxjava.grpc;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Optional.empty;
+
+import com.daml.ledger.javaapi.data.codegen.HasCommands;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * This class should be used to build a valid submission. It provides {@link #create(String, String, List)}
+ * for initial creation and extra builder methods for additional parameters
+ * e.g {@link #withActAs(List)}, {@link #withWorkflowId(String)} e.t.c
+ *
+ * Usage:
+ * <pre>
+ *   var params = CommandsBuilder.create("applicationId", "commandId", commands)
+ *                                   .withAccessToken("token")
+ *                                   .withParty("party")
+ *                                   .with...
+ * <pre/>
+ */
+public class CommandsBuilder {
```

Comment:
Rename to `CommandsSubmission` ? it's not too important that it follows a builder pattern (and there is no build method so it's possibly not really a builder anyway)